consensus/quorum/one-node-one-vote.go

```diff
@@ -109,11 +89,10 @@ func (v *uniformVoteWeight) AmIMemberOfCommitee() bool {
 		return false
 	}
 	identity, _ := pubKeyFunc()
-	everyone := v.DumpParticipants()
+	everyone := v.Participants()
 	for _, key := range identity.PublicKey {
-		myVoterID := key.SerializeToHexStr()
 		for i := range everyone {
-			if everyone[i] == myVoterID {
+			if everyone[i] == key {
```

Comment:
This is comparing pointers. We shouldn't compare pointers. it's error prone.

Reply:
Why, it’s comparing a fixed size buffer in value. , would be point if was plain [] byte. It’s like comparing common.Address, allowed

Reply:
Its shard.BlsPublickKey, not *bls.PublickKey

Reply:
Participants() returns []*bls.PublicKey, pubKeyFunc also returns []*bls.PublicKey. Comparing pointer won't be correct.

Reply:
Ah you're right, my mistake. I forgot to unify this usage to be shard.BlsPublicKey like I did else where, easy fix.